src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java

```diff
@@ -949,6 +941,79 @@ public void testConsecutiveQueries() throws SQLException {
             }
         }
 
+        @Test
+        public void testMaxFetchSize() throws SQLException {
+            try (Connection con = getConnection(); Statement stmt = con.createStatement()) {
+
+                // Default value should return zero
+                long actual = stmt.getMaxFieldSize();
+                assertEquals(actual, (long) 0, "getMaxFieldSize():" + TestResource.getResource("R_incorrectDefault"));
+
+                // Set a new value less than MAX_VALUE, and then get the modified value
+                int newValue = 2012;
+                stmt.setMaxFieldSize(newValue);
+                actual = stmt.getMaxFieldSize();
+                assertEquals(actual, newValue, "MaxFieldSize() : set/get problem");
+
+                // execute a statement with this new max field size
+                stmt.execute("SELECT @@VERSION AS 'SQL Server Version'");
+
+                /*
+                 * Set a new value greater than MAX_VALUE, and then get the modified value SQL Server only supports
+                 * integer limits for setting max rows If the value MAX_VALUE + 1 is accepted, throw exception
+                 */
+                try {
+                    newValue = Integer.MAX_VALUE + 1;
+                    stmt.setMaxFieldSize(newValue);
+                    fail(TestResource.getResource("R_expectedFailPassed"));
+                } catch (Exception e) {
```

Comment:
Catch `SQLServerException` here, to make sure the driver does not throw some other exception.